pandas/core/frame.py

```diff
@@ -10541,25 +10539,22 @@ def _get_data() -> DataFrame:
                 data = self._get_bool_data()
             return data
 
-        numeric_only_bool = com.resolve_numeric_only(numeric_only)
-        if numeric_only is not None or axis == 0:
+        if numeric_only or axis == 0:
```

Comment:
why can't we go through this path unconditionally?

Reply:
Ah, I don't think what I wrote was all too clear, but tried to explain this in https://github.com/pandas-dev/pandas/pull/49551#issuecomment-1304815473. I think we should take this path unconitionality, but there is a behavior change for `numeric_only=False`, `axis=1` and object dtype. I plan to open an issue and do a followup here.

Reply:
Here is an example to make this more explicit (on main):

```
df = pd.DataFrame({'a': [1, 2, 3], 'b': [4, 5, 6]}, dtype=object)
print(df.sum(axis=1, numeric_only=None))
print(df.sum(axis=1, numeric_only=False))
0    5.0
1    7.0
2    9.0
dtype: float64
0    5
1    7
2    9
dtype: object
```

Taking this path unconditionally would be to always get object dtype. I think that's the right thing to do, but would be a change in the default (numeric_only=None) behavior and plan to handle in a follow up.

Reply:
makes sense, thanks

Reply:
I opened #49603